### PR TITLE
Use correct file separator on windows for constants in build_h.jl

### LINKED ISSUE #21330: Base.DATAROOTDIR file separator on windows
Base.DATAROOTDIR  shows up as
` "./share" `  on Windows and not `".\\share"`

### PR DESCRIPTION
**Pull request by musm**
close https://github.com/JuliaLang/julia/issues/21330

### CODE REVIEW

File: test/osutils.jl [hunk 1]
```diff
@@ -26,3 +26,13 @@ end
 @test (@static false && 1) === false
 @test (@static true || 1) === true
 @test (@static false || 1) === 1
+
+# test that path variables use proper path deliminter on windows
```
Comment (tkelman) [inline test/osutils.jl]: delimiter

Comment (musm) [inline test/osutils.jl]: thanks,fixed

**Comment by musm** [comment]
cc @vtjnash  does this like good? Also thanks for the help

**Comment by musm** [comment]
bump

**Comment by musm** [comment]
bump anything preventing this?

**Comment by musm** [comment]
Bump

**Review by nalimilan — APPROVED** [review]
I'm not an expert in Makefiles nor of Windows, but this looks good to me, and since tests passed...

**Comment by musm** [comment]
thanks @nalimilan , is this good to go in this case?

**Pull request merged**